manager/dispatcher/dispatcher.go

```diff
@@ -125,8 +125,18 @@ type clusterUpdate struct {
 
 // Dispatcher is responsible for dispatching tasks and tracking agent health.
 type Dispatcher struct {
-	mu                   sync.Mutex
-	wg                   sync.WaitGroup
+	// Lock to provide  mutually exclusive access to dispatcher fields
+	// e.g. lastSeenManagers, networkBootstrapKeys, lastSeenRootCert etc.
+	// Also used to make atomic the setting of the shutdown flag to 'true' and the
+	// Add() operation on the shutdownWait to make sure that stop() waits for
+	// all operations to finish and disallow new operations from starting.
+	mu sync.Mutex
+	// Flag to indicate shutdown and prevent new operations on the dispatcher.
+	// Set by calling Stop().
+	shutdown bool
```

Comment:
Any reason to avoid using a tombstone channel?

Reply:
Caught up with @stevvooe offline. I agree that using a tombstone is cleaner, but some of the operations don't use a select {} so using a bool saves some code.